File: code/test/apitest/py/autodetectimg.py
#!/usr/bin/env python
#
# Copyright (c) 2005-2009 Jaroslav Gresula
#
# Distributed under the MIT license (See accompanying file
# LICENSE.txt or copy at http://jagpdf.org/LICENSE.txt)
#

import os
import jagpdf
import jag.testlib as testlib

paper = 597.6, 848.68

def image_path(name):
    return os.path.expandvars('${JAG_TEST_RESOURCES_DIR}/images/') + name

def test_main(argv=None):
    doc = testlib.create_test_doc(argv, 'autodetectimg.pdf')
    imgs = []
    imgs.append(doc.image_load_file(image_path('lena.jpg'), jagpdf.IMAGE_FORMAT_JPEG))
    imgs.append(doc.image_load_file(image_path('logo.png'), jagpdf.IMAGE_FORMAT_PNG))
    imgs.append(doc.image_load_file(image_path('lena.jpg'), jagpdf.IMAGE_FORMAT_AUTO))
    imgs.append(doc.image_load_file(image_path('logo.png'), jagpdf.IMAGE_FORMAT_AUTO))
    imgs.append(doc.image_load_file(image_path('lena.jpg')))
    imgs.append(doc.image_load_file(image_path('logo.png')))
    # tbd ~ imgspec
    spec1 = doc.image_definition()
    spec1.file_name(image_path('lena.jpg'))
    spec2 = doc.image_definition()
    spec2.file_name(image_path('logo.png'))
    imgs.append(doc.image_load(spec1))
    imgs.append(doc.image_load(spec2))
    doc.page_start(*paper)
    x, y = 100, 50
    o = 0
    for img in imgs:
        doc.page().canvas().image(img, x + o, y + o)
        o += 30
    doc.page_end()
    doc.finalize()

if __name__ == '__main__':
    test_main()

File: code/test/apitest/cpp/autodetectimg.cpp
// Copyright (c) 2005-2009 Jaroslav Gresula
//
// Distributed under the MIT license (See accompanying file
// LICENSE.txt or copy at http://jagpdf.org/LICENSE.txt)
//


#include <cstdio>
#include <string>
#include <iostream>
#include <stdlib.h>

#include "testcommon.h"
using namespace jag;

// tests that enum default arguments work

namespace
{
  void test_main(int /*argc*/, char** /*argv*/)
  {
      StreamNoop stream;
      pdf::Document doc(pdf::create_stream(&stream));
      doc.page_start(10.0*72, 10.0*72);
      pdf::Canvas canvas(doc.page().canvas());

      std::ostringstream img_path;
      img_path <<  getenv("JAG_TEST_RESOURCES_DIR")
               << "/images/lena.jpg";

      doc.image_load_file(img_path.str().c_str());
      doc.image_load_file(img_path.str().c_str(), pdf::IMAGE_FORMAT_JPEG);

      doc.page_end();
      doc.finalize();
  }
} // anonymous namespace

int autodetectimg(int argc, char ** const argv)
{
    return test_runner(test_main, argc, argv);
}



/** EOF @file */
